Sources/CCryptoBoringSSLShims/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+# This source file is part of the SwiftCrypto open source project
+#
+# Copyright (c) 2021 Apple Inc. and the SwiftCrypto project authors
+# Licensed under Apache License v2.0
+#
+# See LICENSE.txt for license information
+# See CONTRIBUTORS.md for the list of SwiftCrypto project authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+add_library(CCryptoBoringSSLShims STATIC
+  shims.c)
+
+target_include_directories(CCryptoBoringSSLShims PUBLIC
+  include)
+
+target_link_libraries(CCryptoBoringSSLShims PUBLIC
+  CCryptoBoringSSL)
```

Comment:
@compnerd `CCryptoBoringSSLShims` fails to build and I don't know how to fix it:

```
In file included from Sources/CCryptoBoringSSLShims/shims.c:14:
Sources/CCryptoBoringSSLShims/include/CCryptoBoringSSLShims.h:22:10: fatal error: 'CCryptoBoringSSL.h' file not found
#include <CCryptoBoringSSL.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

`CCryptoBoringSSL.h` is in `CCryptoBoringSSL/include`, which we include:

```
target_include_directories(CCryptoBoringSSL PUBLIC
   include)
```

And we link `CCryptoBoringSSL` here.

Are we doing something wrong or something is missing?

Reply:
Hmm, because the include is marked as `PUBLIC` it should propagate to the user, and allow it to find it.

Reply:
#62 has the changes needed for this - feel free to pick up the changes and merge them here.

Reply:
Thanks @compnerd. I tested that it works on Ubuntu 20.04 (`swift:5.3-focal`) (needed https://github.com/apple/swift-crypto/pull/60/commits/5f805719f61810b85370d57aefd42757db4aa725 though):

```
apt-get update && apt-get install -y cmake ninja-build

cmake -G Ninja -D BUILD_SHARED_LIBS=YES -D CMAKE_BUILD_TYPE=Release -B out -S .
ninja
```

On macOS however, we'd need `CryptoKit`:

```
Sources/Crypto/Digests/HashFunctions.swift:18:24: error: cannot find type 'CoreCryptoDigestImpl' in scope
typealias DigestImpl = CoreCryptoDigestImpl
                       ^~~~~~~~~~~~~~~~~~~~
Sources/Crypto/Digests/HashFunctions.swift:30:28: error: cannot find type 'CryptoKit' in scope
    associatedtype Digest: CryptoKit.Digest
```

I tried https://github.com/apple/swift-crypto/pull/60/commits/b20d61504a50b74cd3f3a7df09d5d0203387dead but it's no good.

Reply:
The Darwin handling is supposed to be different because ... well, Apple.  Im not convinced that the CryptoKit variant is correct, but if you say it is, then it might be.  The thing is that `CryptoKit` is a framework, not a library so the linking is different.  However, I did update compnerd/swift-crypto (apple/swift-crypto#62) with the changes to link against CryptoKit).  It doesn't build locally for me due to missing typealiases (maybe I just have an older Xcode).

Reply:
Okay, compnerd/swift-crypto 9989bd67885d6685cebfb2741c19acdb9aba4357 has the changes to build SwiftCrypto on macOS with BoringSSL using the SwiftPM forced build approach.  I do not know what the correct thing is, so you will need to help guide the right solution here.

Reply:
Thanks. I tried 

```
  target_link_options(Crypto PRIVATE
    "$<$<PLATFORM_ID:Darwin>:SHELL:-Xlinker -framework -Xlinker ${CRYPTO_KIT}>")
```

and it didn't work for me either. I got the same errors as [before](https://github.com/apple/swift-crypto/pull/60#discussion_r562444344).

Using `CRYPTO_IN_SWIFTPM_FORCE_BUILD_API` works. I assume this has to do with how `CryptoKit` is used specifically in this library. I tried importing another framework (`Security`) and actually didn't need to do anything extra (i.e., linking).

Reply:
Most of the frameworks on macOS work without the correct dependency tracking due to auto-linking which adds implicit dependencies that the build system doesn't know about.  It is always a good idea to be explicit about it.